### PR TITLE
Issue 9913 pending content page edit mode publish page errors on ng iframe component

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/servlets/LoginEditModeServlet.java [hunk 1]
```diff
@@ -36,12 +36,15 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
...
-					
-				if(_edit_mode_id.getAssetType().equals("contentlet")){
-					com.dotmarketing.portlets.contentlet.model.Contentlet cont =  APILocator.getContentletAPI().findContentletByIdentifier(_edit_mode_id.getId(), false, APILocator.getLanguageAPI().getDefaultLanguage().getId(), APILocator.getUserAPI().getSystemUser(), false);
-					_edit_mode_id.setURI(cont.getMap().get("URL_MAP_FOR_CONTENT").toString());
+
+				if (clickstream != null) {
+					String pageId = clickstream.getLastPageId();
+					_edit_mode_id = (Identifier) HibernateUtil.load(Identifier.class, pageId);
+
+					if (_edit_mode_id.getAssetType().equals("contentlet")) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/servlets/LoginEditModeServlet.java]: constant should be in the left side

File: dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java [hunk 2]
```diff
@@ -4,22 +4,75 @@
...
+    private URLEncoder encoder = new URLEncoder();
+    private static String IN_FRAME_PARAMETER_NAME = "in_frame";
+
     /**
      * 
      */
     private static final long serialVersionUID = 1L;
 
+    protected void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
+        boolean ADMIN_MODE = PageRequestModeUtil.isAdminMode( req.getSession() );
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java]: what is this a constants? shouldn't be adminMode

File: dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java [hunk 3]
```diff
@@ -4,22 +4,75 @@
...
+        boolean notRedirect = notRedirect( req );
+
+        if ( ADMIN_MODE  && !notRedirect ){
+            String pageUrl = req.getAttribute("javax.servlet.forward.request_uri").toString();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java]: This should be a constant

File: dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java [hunk 4]
```diff
@@ -4,22 +4,75 @@
...
+                response.sendRedirect(redirectURL);
+                return;
+            }
+        }
+
+        super.service(req, response);
+    }
+
+    public String getRedirectURL(String pageUrl){
+        return String.format("/dotAdmin/#/c/site-browser?&url=%s", encoder.encode( pageUrl ));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java]: Not sure if this should be configurable, but at least should be a constant easy to indentified

File: dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java [hunk 5]
```diff
@@ -4,22 +4,75 @@
...
+     * <ul>
+     *     <li>The url content the dotAdmin String</li>
+     *     <li>or If teh Referer header content the dotAdmin String</li>
+     * </ul>
+     *
+     * @param req
+     * @return
+     */
+    private boolean notRedirect(HttpServletRequest req) {
+        String refererValue = req.getHeader("Referer");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java]: This should be a global constant (such as Constants interface)

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java [hunk 6]
```diff
@@ -4,22 +4,70 @@
...
+     * @param req
+     * @return
+     */
+    private boolean notRedirect(HttpServletRequest req) {
+        String refererValue = req.getHeader(Constants.REFERER_URL_HTTP_HEADER);
+
+        return (refererValue != null && (refererValue.contains( "host_id=" )
+                    || refererValue.contains( "fromAngular=true" )
+                    || refererValue.contains(PortletURLUtil.URL_ADMIN_PREFIX )
+                    || refererValue.contains( "p_p_id=site-browser" )));
```
Comment (oswaldogallango) [inline dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java]: here the portlet id is hardcoded p_p_id=site-browser should be using the enum

**Review by jgambarios — APPROVED** [review]

**Pull request merged**